tests/typ/compute/construct.typ

```diff
@@ -14,6 +14,27 @@
 #test(rgb("#133337").negate(), rgb(236, 204, 200))
 #test(white.lighten(100%), white)
 
+// Color mixing, in Oklab space by default.
+#test(color.mix(rgb("#ff0000"), rgb("#00ff00")), rgb("#a16400"))
+#test(color.mix(rgb("#ff0000"), rgb("#00ff00"), space: "oklab"), rgb("#a16400"))
+#test(color.mix(rgb("#ff0000"), rgb("#00ff00"), space: "srgb"), rgb("#808000"))
+
+// Slight floating-point error here is acceptable, but it should be basically commutative.
```

Comment:
I think it may not be acceptable. Color mixing usually isn't something you do in a hot code path, so I would rather see a slower implementation that has less rounding errors (or could optimally be proven to never create errors above 1/512 or so). Assuming the current implementation is free of logic errors, this could probably be achieved by using [f128](lib.rs/f128) or fractions (i am not aware of the current dependencies, or whether adding one would be considered acceptable).

Reply:
If we add another dependency like this, it might as well be one that just handles all of the color mixing.

Reply:
Sorry, the comment here is inaccurate — it's just that `color.mix(red, blue, green)` internally mixes `red` with `blue` _and rounds to 8-bit RGBA_ to get an intermediate purple, then mixes 2/3 purple with 1/3 `green` and rounds to 8-bit RGBA again. So the error is due to the rounding to `u8`, and not due to `f64` being insufficiently precise. It didn't seem very important to me because `#5c8169` and `#5c826a` are basically identical to the naked eye, but I can look into eliminating the intermediate rounding.

Reply:
I see there is an [oklab](https://docs.rs/oklab/latest/oklab/) crate, whose source code is making me realize my implementation is lacking a gamma-correction step, oops! So maybe I will switch to that.